docs/cep/tutorials/using_rest_api.md

```diff
@@ -178,234 +178,232 @@ print("\nStream App Deleted: ", result)
+          body: { email, password },
+          method: "POST"
+        })
+        .then(({ jwt, ...data }) => {
+          self._headers.authorization = `bearer ${jwt}`;
+          resolve(data);
+        })
+        .catch(reject);
+    });
+  }
+
+  _handleResponse (response, resolve, reject) {
+    if (response.ok) {
+      resolve(response.json());
+    } else {
+      reject(response);
+    }
+  }
+
+  req (endpoint, { body, ...options } = {}) {
+    const self = this;
+    return new Promise(function (resolve, reject) {
+      fetch(self._url + endpoint, {
+        headers: self._headers,
+        body: body ? JSON.stringify(body) : undefined,
+        ...options,
+      }).then((response) => self._handleResponse(response, resolve, reject));
+    });
+  }
+}
+const email = "nemo@nautilus.com";
+const password = "xxxxxx";
+const federationName = "api-gdn.paas.macrometa.io";
+const federationUrl = `https://${federationName}`;
+
+const isGlobal = false;
+const stream = "tutorialAppInputStream";
+const streamAppName = "stream_app_tutorial";
+const streamAppDef = `@App:name('stream_app_tutorial')
+  @App:description('This application demonstrates how to use user defined functions in the stream app')
+  @App:qlVersion("2")
+  CREATE FUNCTION concatFn[javascript] return string {
+      var str1 = data[0];
+      var str2 = data[1];
+      var str3 = data[2];
+      var response = str1 + str2 + str3;
+      return response;
+  };
+  -- Stream
+  CREATE STREAM tutorialAppInputStream (deviceID string, roomNo int, temperature double);
+  -- Table
+  CREATE TABLE tutorialAppOutputTable (id string, temperature double);
+  @info(name='Query')
+  INSERT INTO tutorialAppOutputTable
+  SELECT concatFn(roomNo,'-',deviceID) as id, temperature
+  FROM tutorialAppInputStream;`;
+
+const run = async function () {
+  try {
+    const connection = new APIRequest(federationUrl);
+
+    /* -------------------- Login (nemo@nautilus.com/xxxxxx) -------------------- */
+
+    const { tenant } = await connection.login(email, password);
+
+    console.log("Log in successful using: ", tenant);
+
+    /* ---------------------------- Create StreamApp ---------------------------- */
+    try {
+      const streamApp = await connection.req("/_fabric/_system/_api/streamapps", {
+        body: {
+          definition: streamAppDef,
+          regions: []
+        },
+        method: "POST"
+      });
+      console.log("STREAM APP CREATED SUCCESSFULLY", streamApp);
+    } catch (e) {
+      if (e.status === 409) {
+        console.log("Stream app already exists, skipping creation");
+      } else {
+        console.log("Error while creating stream app");
+        throw e;
       }
+    }
 
-      _handleResponse(response, resolve, reject) {
-        if (response.ok) {
-          resolve(response.json());
-        } else {
-          reject(response);
-        }
-      }
+    /* --------------------------- Activate StreamApp --------------------------- */
```

Comment:
I am confused. In some places, you use StreamApp, in others "stream app." Why the two spellings?

Reply:
It was already there as part of the code I didn't change these actually

Reply:
done